x-pack/plugins/enterprise_search/server/lib/check_access.ts

```diff
@@ -17,7 +17,7 @@ import { callEnterpriseSearchConfigAPI } from './enterprise_search_config_api';
 
 interface CheckAccess {
   request: KibanaRequest;
-  security: SecurityPluginSetup;
+  security?: SecurityPluginSetup;
```

Comment:
__note:__ The Kibana Security plugin __cannot__ be disabled since 8.0.0. It's the ES Security plugin that can be disabled via `xpack.security.enabled: false`. This means you can safely assume that the `security` dependency is always available. All you need to do is check `security.authz.mode.useRbacForRequest` that depends on security status in ES.

```suggestion
  security: SecurityPluginSetup;
```